### PR TITLE
Examination Regulation UI

### LINKED ISSUE #678: Add view for adding an ExReg
- add a new exam regulation having a name and modules
- creation of new modules

### PR DESCRIPTION
**Pull request by this-kramer**
# Description
- Closes #678 

## Reason for this PR
- We need a view for creating new exam regulations.

## Changes in this PR
- Add view that allows creating exam regulations

### Dependency update
- List all new dependencies (delete if no additions)

## Type of change (remove all that don't apply)
- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Manually tested all use cases this could alter

**Test Configuration**:
- OS: Windows Mac Linux
- Browser: Firefox Chrome Safari Chromium

- Frontend: (remove all that don't apply)
  - [x] Development build
- Backend: (remove all that don't apply)
  - [x] deployed development build

# Checklist: (remove all that don't apply)

- [x] I added corresponding E2E tests (especially for bugfixes)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new linting warnings or console warnings
- [x] I have updated the CHANGELOG.md to include any changes made in this PR (add WIP to the top, if there is none already)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# [v0.14.0](https://github.com/upb-uc4/ui-web/compare/v0.13.1...v0.14.0) (Draft)
```
Comment (tobias-wiese) [inline CHANGELOG.md:1]:
```suggestion
# [v0.13.2 WIP](https://github.com/upb-uc4/ui-web/compare/v0.13.1...v0.14.0)
```
We tag our "next release" as the next subversion of the last sprint increment. E.g. we are in sprint 14 and hat a version v0.13.1, hence the nextr version will be v0.13.2. We will just bump the version on v0.14.0 on the release branch for the sprint increment.

File: src/use/router/index.ts [hunk 2]
```diff
@@ -122,6 +123,15 @@ const router = createRouter({
                 roles: ["Lecturer", "Admin"],
             },
         },
+        {
+            path: "/createExamReg",
```
Comment (tobias-wiese) [inline src/use/router/index.ts:127]:
```suggestion
            path: "/create-exam-reg",
```
Unfortunately, our URLs for the different components have different cases. We are planning to refactor them, s.t. all path are kebab-case. So we should just use kebab-case for the new features, too :)

File: src/use/router/index.ts [hunk 3]
```diff
@@ -122,6 +123,15 @@ const router = createRouter({
...
+            name: "examRegForm",
+            component: ExamRegForm,
+            meta: {
+                title: "Examination Registration Creation" + suffix,
```
Comment (tobias-wiese) [inline src/use/router/index.ts:131]:
```suggestion
                title: "Examination Regulation Creation" + suffix,
```
Note: The title is used for displaying something as the tab title in the browser. "Examination Regulation Creation" could be a bit long, we could use the short form "Exam" like in the navbar.

File: src/views/admin/CreateExamRegForm.vue [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+    export default {
+        name: "CreateExamRegForm",
+        components: {
+            ExRegModuleSection,
+            UnsavedChangesModal,
+            ExRegInfoSection,
+            LoadingComponent,
+        },
+        setup: function () {
```
Comment (tobias-wiese) [inline src/views/admin/CreateExamRegForm.vue:67]:
```suggestion
        setup() {
```
Just a feeling of keeping the files consistent :)

File: tests/e2e/navbarDesktop.spec.ts [hunk 5]
```diff
@@ -49,6 +49,13 @@ describe("admin", () => {
         cy.get("#nav_desktop_admin_menu_courses_create").click();
         cy.url().should("contain", "createCourse");
     });
+
+    it("exam-regs", () => {
+        // create exam reg form
+        cy.get("#nav_desktop_admin_menu_exreg").trigger("mouseover");
+        cy.get("#nav_desktop_admin_menu_exreg_create").click();
+        cy.url().should("contain", "createExamReg");
```
Comment (tobias-wiese) [inline tests/e2e/navbarDesktop.spec.ts:57]:
```suggestion
        cy.url().should("contain", "create-exam-reg");
```
Just commit this when committing the path change in the router, too

File: tests/e2e/navbarMobile.spec.ts [hunk 6]
```diff
@@ -83,6 +83,15 @@ describe("admin", () => {
...
         cy.url().should("contain", "createCourse");
     });
+
+    it("exam-regs", () => {
+        cy.get(navbar_burger_menu).click();
+
+        // create exam regulation
+        cy.get("#nav_mobile_menu_exreg").click();
+        cy.get("#nav_mobile_admin_menu_exreg_create").click();
+        cy.url().should("contain", "createExamReg");
```
Comment (tobias-wiese) [inline tests/e2e/navbarMobile.spec.ts:93]:
```suggestion
        cy.url().should("contain", "create-exam-reg");
```
see above

File: src/components/exreg/ExRegModuleSection.vue [hunk 7]
```diff
@@ -0,0 +1,151 @@
+<template>
+    <section class="border-t-2 py-8 border-gray-400">
+        <div class="lg:flex">
+            <div class="w-full lg:w-1/3 lg:block mr-12 flex flex-col mb-4">
+                <label class="block text-gray-700 text-md font-medium mb-2"> Modules </label>
+                <label class="block text-gray-600"> Add modules to the examination regulation </label>
```
Comment (tobias-wiese) [inline src/components/exreg/ExRegModuleSection.vue:6]:
```suggestion
                <label class="block text-gray-600"> Add modules to the examination regulation.</label>
```

File: src/components/exreg/ExRegModuleSection.vue [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+                            id="moduleName"
+                            v-model.trim="moduleNameInput"
+                            class="w-full form-input input-text"
+                            placeholder="Module Name"
+                        />
+                    </div>
+                    <div v-else-if="moduleUsed" class="mb-4">
+                        <label class="text-gray-700 text-md font-medium my-3">
+                            <i class="text-red-400 fas fa-times mr-2"></i>
+                            Module {{ moduleID }} already selected!
```
Comment (tobias-wiese) [inline src/components/exreg/ExRegModuleSection.vue:26]:
```suggestion
                            Module '{{ moduleID }}' already selected!
```

File: src/views/admin/CreateExamRegForm.vue [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+                            Create Examination Regulation
+                        </button>
+                    </div>
+                </div>
+
+                <!-- different button layout for mobile -->
+                <div class="sm:hidden">
+                    <button id="mobileCancel" type="button" class="mb-4 w-full btn btn-blue-secondary" @click="back">Cancel</button>
+                    <button id="mobileCreateCourse" :disabled="!canCreate" class="mb-4 w-full btn btn-blue-primary" @click="createExamReg">
+                        Create Examination Regulation
```
Comment (tobias-wiese) [inline src/views/admin/CreateExamRegForm.vue:36]:
```suggestion
                        Create Exam Regulation
```
In the navbar you use the short form "exam". We should use this consistently.

File: src/views/admin/CreateExamRegForm.vue [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    export default {
+        name: "CreateExamRegForm",
+        components: {
+            ExRegModuleSection,
+            UnsavedChangesModal,
+            ExRegInfoSection,
+            LoadingComponent,
+        },
+        setup: function () {
+            const heading = "Create Examination Regulation";
```
Comment (tobias-wiese) [inline src/views/admin/CreateExamRegForm.vue:68]:
```suggestion
            const heading = "Create Exam Regulation";
```
see above

File: src/views/admin/CreateExamRegForm.vue [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+                const handledResponse = validationResponseHandler.handleResponse(response);
+                if (handledResponse) {
+                    const toast = useToast();
+                    toast.success(`Examination regulation ${examRegName.value} created.`);
+                    // reset ui, add new exam regs and modules since blockchain is lazy...
+                    existingExamRegNames.value.push(examRegName.value);
+                    let newModules = selectedModules.value.filter((m) => !existingModules.value.find((x) => m == x));
+                    existingModules.value.push(...newModules);
+                    selectedModules.value = [] as Module[];
+                    console.log(selectedModules.value);
```
Comment (tobias-wiese) [inline src/views/admin/CreateExamRegForm.vue:124]:
```suggestion
```

**Review by tobias-wiese — APPROVED** [review]
Nice work! I just commented some minor things, nothing functional :)

**Review by theovier — APPROVED** [review]

**Pull request merged**